Server.Core/RequestBuilder.cs

```diff
@@ -0,0 +1,122 @@
+﻿ using System;
+ using System.Text;
+
+namespace Server.Core
+{
+    public class RequestBuilder
+    {
+        private Request request;
+
+        public RequestBuilder()
+        {
+            request = new Request();
+        }
+
+        public Request BuildRequest()
+        {
+            return request;
+        }
+
+        public Request BuildRequestFromWholeMessage(byte[] requestMessage)
+        {
+            if (requestMessage.Length == 0)
+            {
+                SetRequestToNoValues();
+            }
+            else
+            {
+                ParseMessageBytes(requestMessage);
+            }
+            return request;
+        }
+
+        public void SetRequestHttpVersion(string httpVersion)
```

Comment:
Is having `Request` in the name of all of these methods necessary. It *is* the `RequestBuilder` after all.